### PR TITLE
Sri hari: Neetcode-150/articles

### PR DESCRIPTION
**Pull request by Srihari2222**
@neetcode-gh 
- Took 5 hours for this work.

---
- There is no Deque() ds in the neetcode.io, cross check this as sliding window maximum problem needs deque. 
- I written the deque() code based on your provided heap, queue links.

### CODE REVIEW

File: articles/sliding-window-maximum.md [hunk 1]
```diff
@@ -0,0 +1,899 @@
...
+            }
+            r++;
+        }
+
+        return output;
+    }
+};
+```
+
+```javascript
```
Comment (neetcode-gh) [inline articles/sliding-window-maximum.md:829]: Good catch. I will add the deque class to the environment shortly. I made some minor corrections to your JS deque solution. Please update it and then I will merge this PR:

---

class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    maxSlidingWindow(nums, k) {
        const n = nums.length;
        const output = new Array(n - k + 1);
        const q = new Deque();
        let l = 0, r = 0;

        while (r < n) {
            while (q.size() && nums[q.back()] < nums[r]) {
                q.popBack();
            }
            q.pushBack(r);

            if (l > q.front()) {
                q.popFront();
            }

            if ((r + 1) >= k) {
                output[l] = nums[q.front()];
                l++;
            }
            r++;
        }

        return output;
    }
}

**Comment by neetcode-gh** [comment]
Fantastic work @Srihari2222

I just had one comment, the rest looks good to me.

**Comment by neetcode-gh** [comment]
And just to clarify this is the Deque library we will use for JS going forward: https://github.com/datastructures-js/deque/

**Comment by neetcode-gh** [comment]
Thanks!

**Comment by Srihari2222** [comment]
@neetcode-gh 
Thanks, I changed the deque code. You can merge this PR. There is one more PR that is active.

**Pull request merged**